Mathlib/Order/Filter/Cocardinal.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2024 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+import Mathlib.Order.Filter.CountableInter
+import Mathlib.Order.Filter.CardinalInter
+import Mathlib.SetTheory.Cardinal.Ordinal
+import Mathlib.SetTheory.Cardinal.Cofinality
+import Mathlib.Order.Filter.Bases
+
+/-!
+# The cocardinal filter
+
+In this file we define `Filter.cocardinal hc`: the filter of sets with cardinality less than
+  a regular cardinal `c` that satisfies `Cardinal.aleph0 < c`.
+  Such filters are `CardinalInterFilter` with cardinality `c`.
+
+-/
+
+open Set Filter Cardinal
+
+universe u
+variable {ι : Type u} {α β : Type u}
+variable {c : Cardinal.{u}} {hreg : Cardinal.IsRegular c}
+variable {l : Filter α}
+
+namespace Filter
+
```

Comment:
I've resolved all other comments, but can you comment on why you would like to do this? This would mean that you need to make the dependence on α explicit much more often, right?

Reply:
I suggested this because otherwise the expression `cocardinal hreg` contains no information from which we can deduce `α`. Indeed if you test the following:
```lean
variable (hreg : c.IsRegular)
#check cocardinal hreg
```
you will see that the output is `cocardinal hreg : Filter ?m.990` indicating that `α` is a metavariable.

So I think you should make this change.